tower.js

```diff
@@ -497,4 +490,31 @@ CollectorTower.prototype.extractResource = function(resource, state){
     state.uranium += this.extractRate;
   }
   resource.extract(this.extractRate);
+}
+
+
+function getNewTower(type, pos, state){
+  switch(type){
```

Comment:
hm should have added me as a reviewer lol
this should be a map object and not a big case/switch it is making me uncomfortable :(

Reply:
storing the name of the object as a string and then using that string to find the object later ???? what is this

Reply:
i'm going to report you to liberatore and he will retroactively fail you for this,

Reply:
Most of the changes in this PR were from 2020. A lot of refactoring is needed at some point for this project.